dcos/dcos-cluster-one-master.json

```diff
@@ -699,48 +744,36 @@
       "type": "Microsoft.Storage/storageAccounts"
     }, 
     {
-      "apiVersion": "[variables('apiVersionDefault')]", 
+      "apiVersion": "2016-07-01",
+      "name": "[variables('virtualNetworkName')]",
+      "type": "Microsoft.Resources/deployments", 
       "dependsOn": [
         "[concat('Microsoft.Network/networkSecurityGroups/', variables('agentPublicNSGName'))]", 
         "[concat('Microsoft.Network/networkSecurityGroups/', variables('agentPrivateNSGName'))]"
       ], 
-      "location": "[resourceGroup().location]", 
-      "name": "[variables('virtualNetworkName')]", 
-      "properties": {
-        "addressSpace": {
-          "addressPrefixes": [
-            "[variables('masterAddressPrefix')]", 
-            "[variables('agentAddressPrefix')]"
-          ]
-        }, 
-        "subnets": [
-          {
-            "name": "[variables('masterSubnetName')]", 
-            "properties": {
-              "addressPrefix": "[variables('masterAddressPrefix')]"
-            }
-          }, 
-          {
-            "name": "[variables('agentPublicSubnetName')]", 
-            "properties": {
-              "addressPrefix": "[variables('agentPublicAddressPrefix')]", 
-              "networkSecurityGroup": {
-                "id": "[resourceId('Microsoft.Network/networkSecurityGroups', variables('agentPublicNSGName'))]"
-              }
-            }
-          }, 
-          {
-            "name": "[variables('agentPrivateSubnetName')]", 
-            "properties": {
-              "addressPrefix": "[variables('agentPrivateAddressPrefix')]", 
-              "networkSecurityGroup": {
-                "id": "[resourceId('Microsoft.Network/networkSecurityGroups', variables('agentPrivateNSGName'))]"
-              }
-            }
-          }
-        ]
-      }, 
-      "type": "Microsoft.Network/virtualNetworks"
+      "properties": { 
+        "mode": "incremental", 
+        "templateLink": {
+          "uri": "[concat('https://raw.githubusercontent.com/brendandburns/acs/master/dcos/dcos-', parameters('virtualNetworkNewOrExisting'), '-vnet.json')]",
```

Comment:
1. this is referring to brendanburns github
2. Can we keep this as one file?  Our original templates were in multiple parts.  This lead to 2 problems: debugging and surfacing errors to users became difficult, and this also introduced a new global point of failure (GitHub or wherever we host the file).  One thought we had was to auto generate a main file from parts so that we can achieve the modularity, but also have a single file to pass to ARM.